src/ACEtk/block/SecondaryParticleEnergyDistributionBlock/src/ctor.hpp

```diff
@@ -0,0 +1,101 @@
+SecondaryParticleEnergyDistributionBlock() = default;
+
+SecondaryParticleEnergyDistributionBlock(
+    const SecondaryParticleEnergyDistributionBlock& base ) :
+  BaseBlockWithLocators( base ),
+  tyrh_( base.tyrh_ ),
+  frames_( base.frames_ ) {
+
+  if ( Base::owner() ) {
+
+    this->frames_.clear();
+    this->generateFrames();
+  }
+}
+
+SecondaryParticleEnergyDistributionBlock(
+    SecondaryParticleEnergyDistributionBlock&& base ) :
+  BaseBlockWithLocators( std::move( base ) ),
+  tyrh_( std::move( base.tyrh_ ) ),
+  frames_( std::move( base.frames_ ) ) {
+
+  if ( Base::owner() ) {
+
+    this->frames_.clear();
+    this->generateFrames();
+  }
+}
+
+/**
+ *  @brief Constructor (without reference frames for DNED)
+ *
+ *  @param[in] distributions    the energy distribution data
+ */
+SecondaryParticleEnergyDistributionBlock(
+    std::vector< EnergyDistributionData > distributions ) :
+  BaseBlockWithLocators( "DLWH", std::move( distributions ) ),
+  tyrh_( std::nullopt ) {
+
+  this->generateFrames();
+}
+
+/**
+ *  @brief Constructor (with reference frames for DLWH)
+ *
+ *  @param[in] distributions    the energy distribution data
+ *  @param[in] frames           the reference frames
+ */
+SecondaryParticleEnergyDistributionBlock(
+    std::vector< EnergyDistributionData > distributions,
+    std::vector< ReferenceFrame > frames ) :
+  BaseBlockWithLocators( "DLWH", std::move( distributions ) ),
+  tyrh_( std::move( frames ) ) {
+
+  this->generateFrames();
+}
+
+/**
+ *  @brief Constructor (without reference frames for DNED)
+ *
+ *  @param[in] ldlw    the begin iterator of the LDLW block in the XSS array
+ *  @param[in] dlw     the begin iterator of the DLW block in the XSS array
+ *  @param[in] end     the end iterator of the DLW block in the XSS array
+ *  @param[in] nr      the number of reactions
+ */
+SecondaryParticleEnergyDistributionBlock(
+    Iterator lsig, Iterator sig, Iterator end, unsigned int nr ) :
+  BaseBlockWithLocators( "DLWH", lsig, sig, end, nr ),
+  tyrh_( std::nullopt ) {
+
+  this->generateFrames();
+}
+
+/**
+ *  @brief Constructor (with reference frames for DLWH)
```

Comment:
Should this be made consistent with line 58? If so, repeat for line 43.

Reply:
Whoever wrote this code needs to do a better job.